src/bootstrap/dependency-validator.ts

```diff
@@ -270,23 +271,40 @@ export class DependencyPropValidator {
   private infoResolver : { [type in ModuleType] :
     (name : string, packageName ?: string) => FunctionInfoType } = {
     "internal": (name) => {
-      return this.internalManager.infoMap.get(name);
+      // Internals does not require complete Meta-Function
+      return this.internalManager.infoMap.get(name) as MetaFunction;
     },
     "external": (name, packageName) => {
       const externalInfo = this.externalManager.getFunctionInfo(name, packageName);
       return externalInfo;
     },
     "bop": (name) => {
-      return this.systemConfig.businessOperations.find(bop => bop.name === name);
+      const bopDetails = this.systemConfig.businessOperations.find(bop => bop.name === name);
+
+      return {
+        "input": bopDetails.input,
+        "output": bopDetails.output,
+        "functionName": name,
+        "entrypoint": "",
+        "description": "",
```

Comment:
After seeing this I realised it might be a good idea to add an optional description field to BOps.

Reply:
I definitely agree on this. Perhaps as an optional field.